src/app/jscaip/tests/RulesUtils.spec.ts

```diff
@@ -78,24 +83,37 @@ export class RulesUtils {
                 .withContext(minimax.name + ' should consider it a draw').toBe(0);
         }
     }
-    public static expectSecondStateToBeBetterThanFirst(weakerState: AbstractGameState,
-                                                       weakMove: Move,
-                                                       strongerState: AbstractGameState,
-                                                       strongMove: Move,
-                                                       minimax: Minimax<Move, AbstractGameState>)
+    public static expectSecondStateToBeBetterThanFirst<M extends Move, S extends GameState, L>(
+        minimax: Minimax<M, S, L>,
+        weakerState: S,
+        weakMove: MGPOptional<M>,
+        strongerState: S,
+        strongMove: MGPOptional<M>)
     : void
     {
-        const weakValue: number = minimax.getBoardValue(new MGPNode(null, weakMove, weakerState)).value;
-        const strongValue: number = minimax.getBoardValue(new MGPNode(null, strongMove, strongerState)).value;
-        expect(weakValue).toBeLessThan(strongValue);
+        const weakValue: number =
+            minimax.getBoardValue(new MGPNode(weakerState, MGPOptional.empty(), weakMove)).value;
+        const strongValue: number =
+            minimax.getBoardValue(new MGPNode(strongerState, MGPOptional.empty(), strongMove)).value;
+        expect(weakerState.getCurrentPlayer())
+            .withContext('expectSecondStateToBeBetterThanFirst expects states for the same player')
```

Comment:
why ? should not matter.

Reply:
est-ce qu'on veut vraiment comparer des states de joueurs différents ? Ou justement est-ce que ça serait pas des erreurs/typos si on le faisait ?
Si on souhaite vraiment le faire pour une raison sensée, je propose d'enlever cet expect et de comparer les valeurs après multiplication par leur modificateurs

Reply:
Si à un moment on a une IA à qui ont veut apprendre que mieux vaut perdre au tour 12 qu'au tour 10, il faudra pouvoir faire cette comparaison !
Donc oui, on veut !

Reply:
tour 12 et 10 c'est le même joueur pourtant ? :D  Mais changé

Reply:
On ferais pas plutôt expectSecondStateToBeBetterThanFirst**For** et rajouter Player dans les arguments et faire adapter uniquement à player plutôt que juger en fonction du "propriétaire" du state pour weaker et pour stronger ?

Reply:
yep, done pour le "...For" !